UI/setup/upgrade_info.html

```diff
@@ -50,14 +65,29 @@
                             options = ar_accounts
                             value_attr = 'accno'
                             text_attr = 'desc'
+                            disabled = AR_DISABLED
 } ?>
 </div>
 <div class="input_row">
 <p>
-  LedgerSMB supports multiple <em>Accounts payable (AP)</em> accounts
-per company. One of those must be the system default. Please select
-your default below in case of multiple. If the list below is empty,
-your database is in an inconsistent state and needs to be fixed first.
+  LedgerSMB supports multiple <em>Accounts payable (AP) </em> accounts
+per company.
+  <?lsmb IF ap_accounts.size > 1 ?>
+    <span>One of those must be the system default.
+   Please select your default below.</span>
+  <?lsmb ELSE; ?>
+    <span>Your AP account has been selected to be the
+    system default. Please review.</span>
+  <?lsmb END ?>
+  <?lsmb AP_DISABLED = "";
+    IF ap_accounts.size <= 1 && !ap_accounts.0.id; ?>
+    <span style='font-weight:bold'>
+          No <em>AP</em> account available, your database is
+          in an inconsistent state and needs to be fixed first.
```

Comment:
Are we 100% sure that it's impossible to run books without an AR or AP account? I was under the impression we can (and thus should allow migration of data which doesn't).

Reply:
Hmmm...  I can imagine some financial services might be able to work without an AR account because clients might actually be vendors that one accepts money on behalf of, and then distribute on a discounted basis.

At the same time, you *could* do everything over GL and not worry bout the map fields at all.

To improve this check maybe one should check to see if an AR account exists and, if not, if there are records in the ar table?  Ditto for AP?

Reply:
In order to accept no AR account, there must not be any customer defined, and no vendor for AP. Migration will fail otherwise because we explicitly check for existing entity credit/debit account.

Reply:
Ok. Then the direction proposed by Chris sounds like the right way to go: no customer ecas *and* no AR? Then no problem. Same for AP.

Reply:
Done. Added Dojo form validation and submit button enabling